src/scores/probability/fss_impl.py

```diff
@@ -0,0 +1,528 @@
+"""
+This module contains methods related to the FSS score
+
+For an explanation of the FSS, and implementation considerations,
+see: [Fast calculation of the Fractions Skill Score][fss_ref]
+
+[fss_ref]:
+https://www.researchgate.net/publication/269222763_Fast_calculation_of_the_Fractions_Skill_Score
+"""
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import TYPE_CHECKING, Iterable, Optional, Tuple, Union
+
+import numpy as np
+import numpy.typing as npt
+import xarray as xr
+
+from scores import utils
+from scores.typing import XarrayLike
+
+# Note: soft keyword `type` only support from >=3.10
+# "tuple" of 64 bit floats
+f8x3 = np.dtype("f8, f8, f8")
+# Note: `TypeAlias` on variables for python <=3.9
+if TYPE_CHECKING:  # pragma: no cover
+    FssDecomposed = Union[np.ArrayLike, np.DtypeLike]
+else:  # pragma: no cover
+    FssDecomposed = npt.NDArray[f8x3]
+
+
+class DimensionError(ValueError):
+    """
+    Wrapper for specific coordinate/dimension mismatch errors
+    """
+
+    # better to be explicit because python doesn't have braces for scoping its
+    # control flow
+    pass  # pylint: disable=unnecessary-pass
+
+
+class FssComputeMethod(Enum):
+    """
+    Choice of compute backend for FSS.
+    - Standard
+        - NUMPY (default)
+    - Optimized
```

Comment:
If only one thing is available, strip out the other options until such time as we want to include them.

Reply:
I kept it as a scaffolding in case others want to implement particular methods. But I can remove the unimplemented ones.

Reply:
Will move all backends to `scores/fast/fss` and preserve the choice there.

The exposed method(s) in `scores/continuous/fss` will only contain the default `numpy` backend for now.

Reply:
I've removed most of the other methods except numba, just so the example logic on how to check for compatiblity is there in `scores.fast.fss`